exporters/variables-scss/src/generators/fileGenerator.ts

```diff
@@ -18,16 +27,46 @@ export const generateFiles = (
   });
 };
 
-export const generateIndexFile = (files: { fileName: string; content: string }[]) => {
+export const generateIndexFile = (files: { fileName: string; content: string }[], hasJsOutput: boolean = false) => {
   return `${files
     .map((file) => {
-      const baseName = file.fileName.replace(/^_/, '').replace(/\.scss$/, '');
+      const fileExtension = hasJsOutput ? 'ts' : 'scss';
+      const baseName = file.fileName.replace(/^_/, '').replace(new RegExp(`\\.${fileExtension}$`), '');
 
-      return `@forward '${baseName}';`;
+      return hasJsOutput ? `export * from './${baseName}';` : `@forward '${baseName}';`;
     })
     .join('\n')}\n`;
 };
 
+export const generateRootThemesFileImports = (themes: TokenTheme[], hasJsOutput: boolean): string => {
+  return themes
+    .map((theme) => {
+      return hasJsOutput
+        ? `import * as ${toCamelCase(theme.name)} from './${THEMES_DIRECTORY}/${theme.name}';`
+        : `@use '${THEMES_DIRECTORY}/${theme.name}';`;
+    })
+    .join('\n');
+};
+
+export const generateRootThemesFileContent = (themes: TokenTheme[], hasJsOutput: boolean): string => {
```

Comment:
I think you can refactor this like that the templating function will be passed as a callback. That means, that one function will only handle mapping to the callback and the other two will handle the templating.

```js
const jsImportStatement = (name: string) => `import * as ${toCamelCase(name)} from './${THEMES_DIRECTORY}/${name}';`;

const scssImportStatement = (name: string) => `@use '${THEMES_DIRECTORY}/${name}';`;

const prepareImportStatementCallback = (hasJsOutput: boolean) => (theme) => hasJsOutput ? jsImportStatement(theme.name) : scssImportStatement(theme.name);

const generateRootThemesFileImports = (themes: TokenTheme[], hasJsOutput: boolean): string => themes.map(prepareImportStatementCallback(hasJsOutput));
```

Or something like that.